src/components/QuestDetailsWrapper/PlayStreakQuest.stories.tsx

```diff
@@ -535,3 +538,101 @@ export const ActiveWalletSwitchWalletExistingWalletSkipSignature: Story = {
     )
   }
 }
+
+const windowEth = new InjectedProviderMock()
+
+const logErrorMock = fn()
+const trackEventMock = fn()
+
+export const TestSwitchToChainNoEIP3085: Story = {
+  args: {
+    ...mockProps
+  },
+  decorators: [
+    (Story) => {
+      window.ethereum = windowEth
+      return <Story />
+    }
+  ],
+  render: (args) => {
+    const activeWallet = window.ethereum.address
+    const { address } = useAccount()
+    return (
+      <QuestDetailsWrapper
+        {...args}
+        getActiveWallet={async () => Promise.resolve(activeWallet)}
+        getGameplayWallets={async () => [
+          { id: 1, wallet_address: address ?? '' }
+        ]}
+        updateActiveWallet={async () => {
+          console.log('updateActiveWallet', address)
+        }}
+        getUserPlayStreak={async (): Promise<UserPlayStreak> => {
+          return {
+            current_playstreak_in_days: 5,
+            completed_counter: 3,
+            accumulated_playtime_today_in_seconds: 1800,
+            last_play_session_completed_datetime: new Date().toISOString()
+          }
+        }}
+        logError={(...args) => {
+          console.error('quest log error: ', ...args)
+          logErrorMock(...args)
+        }}
+        getQuest={async () => {
+          const mockQuestOneApeChainReward: Quest = {
+            ...mockQuest,
+            rewards: [
+              {
+                id: 1,
+                name: 'Some ApeChain Reward',
+                contract_address: '0xb85Df74eB6db8C2D87c3bD7d4Ee1A27929643dA3',
+                decimals: 18,
+                image_url:
+                  'https://gateway.valist.io/ipfs/bafkreicwp22quggyljn3b4km2we2asaq256yyfa2qyxrapu7qnuasbbnrq',
+                token_ids: [],
+                numClaimsLeft: '2357',
+                amount_per_user: 200000000000000000000000,
+                chain_id: 33139,
+                reward_type: 'ERC20',
+                marketplace_url: 'https://hyperplay.xyz'
+              }
+            ]
+          }
+          return mockQuestOneApeChainReward
+        }}
+        trackEvent={(...args) => {
+          console.log('tracking this event ', ...args)
+          trackEventMock(...args)
+        }}
+      />
+    )
+  },
+  play: async ({ canvasElement }) => {
+    await wait(100)
+    const canvas = within(canvasElement)
+    const claimButton = canvas.getByRole('button', { name: /Claim/i })
+    claimButton.click()
+    await wait(100)
+    const confirmButton = canvas.getByRole('button', { name: /Confirm/i })
+    confirmButton.click()
+    await wait(100)
+    await expect(logErrorMock).toBeCalled()
+    await wait(100)
+    const errorBanner = canvas.getByText(
+      'Please switch to ApeChain within your wallet, or try again with MetaMask.'
+    )
+    expect(errorBanner).toBeInTheDocument()
```

Comment:
this logic is correct but we don't need these hardcoded waits, you can use `waitFor`

Reply:
thx for helping make the changes